CONTRIBUTING.md

```diff
@@ -123,6 +120,24 @@ To add a test to the Flutter package, create a file whose name
 ends with `_test.dart` in the `packages/flutter/test` directory. The
 test should have a `main` function and use the `test` package.
 
+Working with flutter tools
+--------------------------
+
+The flutter shell tools itself is built when you run `flutter` for the first time and each time
+you run `flutter upgrade`. If you want to alter and re-test the shell tool's behavior itself,
+delete `bin/cache/flutter_tools.*` so that the shell tool will be rebuilt from Dart sources
+in `packages/flutter_tools` the next time you run `flutter`.
+
+flutter_tools' tests run inside the dart command line VM rather than recursively in the
+flutter shell. To run the test:
+
+* `cd packages/flutter_tools`
+* `dart test/all.dart`
+
+The pre-built shell tool runs with the observatory (thus debugging) off by default.
```

Comment:
It's not that it runs with observatory turned off so much as it runs in release mode (not checked mode), and in checked mode asserts and the observatory service and so forth are turned off.